modules/transport-grpc/src/main/java/org/opensearch/transport/grpc/proto/response/common/StructProtoUtils.java

```diff
@@ -33,38 +33,40 @@ private StructProtoUtils() {
     public static Value toProto(Object javaObject) {
         Value.Builder valueBuilder = Value.newBuilder();
 
-        if (javaObject == null) {
-            // Null
-            valueBuilder.setNullValue(NullValue.NULL_VALUE);
-        } else if (javaObject instanceof Number) {
-            // Number - use doubleValue() to handle all numeric types
-            valueBuilder.setNumberValue(((Number) javaObject).doubleValue());
-        } else if (javaObject instanceof String) {
-            // String
-            valueBuilder.setStringValue((String) javaObject);
-        } else if (javaObject instanceof Boolean) {
-            // Boolean
-            valueBuilder.setBoolValue((Boolean) javaObject);
-        } else if (javaObject instanceof List) {
-            // List
-            ListValue.Builder listBuilder = ListValue.newBuilder();
-            for (Object listEntry : (List) javaObject) {
-                listBuilder.addValues(toProto(listEntry));
+        switch (javaObject) {
+            case null ->
+                // Null
+                valueBuilder.setNullValue(NullValue.NULL_VALUE);
+            case Number number ->
+                // Number - use doubleValue() to handle all numeric types
+                valueBuilder.setNumberValue(number.doubleValue());
+            case String string ->
+                // String
+                valueBuilder.setStringValue(string);
+            case Boolean bool ->
+                // Boolean
+                valueBuilder.setBoolValue(bool);
+            case List<?> list -> {
+                // List
```

Comment:
nit: some of the one line comments seem not to be adding any value, can remove them?